tango_simlib/tango_launcher.py

```diff
@@ -31,37 +31,42 @@
 required_argument('--server-command', help="TANGO server executable command")
 required_argument('--server-instance', help="TANGO server instance name")
 required_argument('--port', help="TCP port where TANGO server should listen")
+parser.add_argument('--file', help="ASCII file containing device configuration parameters" +
+                    "and device network access parameter")
 parser.add_argument('--put-device-property', action='append', help=
                     "Put a device property into the TANGO DB, format is: "
                     "'device/name/X:property_name:property_value'. Only allows "
                     "properties to be set on devices started with this command. "
                     "Can be specified multiple times.",
                     dest='device_properties', default=[])
 
-def register_device(name, device_class, server_name, instance):
-    dev_info = tango.DbDevInfo()
-    dev_info.name = name
-    dev_info._class = device_class
-    dev_info.server = "{}/{}".format(server_name.split('.')[0], instance)
-    print """Attempting to register TANGO device {!r}
-    class: {!r}  server: {!r}.""".format(
+def register_device(name, device_class, server_name, instance, file=None):
+    if file:
+        db = tango.Database(file)
+    else:
+        dev_info = tango.DbDevInfo()
+        dev_info.name = name
+        dev_info._class = device_class
+        dev_info.server = "{}/{}".format(server_name, instance)
+        print """Attempting to register TANGO device {!r}
+        class: {!r}  server: {!r}.""".format(
             dev_info.name, dev_info._class, dev_info.server)
-    db = tango.Database()
-    db.add_device(dev_info)
+        db = tango.Database()
+        db.add_device(dev_info)
 
-def put_device_property(dev_name, property_name, property_value):
-    db = tango.Database()
+def put_device_property(dev_name, property_name, property_value, file):
```

Comment:
`file` is a reserved python keyword, and we are not allowed to use it as a variable name.

Reply:
`file` is an optional commad-line argument, so it would be better to make it a keyword argument here. You also need to check if a file path has been provided before you instantiate a `Database` object.

Reply:
I don't think you need to make the parameter `file` a keyword argument here. The function will be called with all arguments even if the user is not using the file as a database. `opts` will have a default value `None` set for `opts.file`.
You can change it on the `def register_device(...` function.